src/components/Chips/index.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { ReactElement } from 'react'
+import { Icon, Text } from '../index'
+
+import { TChipsProps } from './types'
+import classNames from 'classnames'
+import '../../assets/styles/components/_chips.scss'
+
+export const Chips = (props: TChipsProps): ReactElement => {
+  const {
+    type = 'filled',
+    color = 'primary',
+    size = 'large',
+    text,
+    disabled,
+    className = '',
+    leftIconProps,
+    withAction,
+    onClick
+  } = props
+
+  return (
+    <div
+      className={classNames(
+        `chips chips--${type} chips--${disabled ? 'disabled' : color} chips--${size}`,
+        className
+      )}
+    >
+      {leftIconProps?.name && (
+        <Icon
+          size={size == 'small' ? 'xsmall' : 'small'}
+          type={disabled ? 'disabled' : type == 'filled' ? 'inverse' : color}
```

Comment:
you can create a variable like this `
const customType =  disabled ? 'disabled' : type == 'filled' ? 'inverse' : color;
and then reuse it instead of duplications